packages/app/components/stack-modal/StackModal.tsx

```diff
@@ -32,100 +31,226 @@ import {
-          </StackDetail>
-        </div>
-        <div className="w-full my-4 border-b border-surface-50"></div>
-        <div className="px-4">
-          <TitleText size={2} weight="bold">
-            Orders
-          </TitleText>
-          <StackProgress stackOrder={stackOrder} />
-          <StackInfo stackOrder={stackOrder} />
-          {totalStackOrdersDone(stackOrder) > 0 && (
-            <StackOrdersTable stackOrder={stackOrder} />
+          <div className="w-full my-4 border-b border-surface-50"></div>
+          <div className="px-4">
+            <TitleText size={2} weight="bold">
+              Orders
+            </TitleText>
+            <StackProgress stackOrder={stackOrder} />
+            <StackInfo stackOrder={stackOrder} />
+            {totalStackOrdersDone(stackOrder) > 0 && (
+              <StackOrdersTable stackOrder={stackOrder} />
+            )}
+          </div>
+        </ModalContent>
+        <ModalFooter>
+          {!stackOrder.cancelledAt && !stackIsComplete && (
+            <Button
+              size="sm"
+              action="secondary"
+              onClick={() => setIsOpenCancellationDialog(true)}
+              width="full"
+            >
+              Cancel Stacking
+            </Button>
           )}
-        </div>
-      </ModalContent>
-      <ModalFooter>
-        {!stackIsComplete && (
-          <Button
-            size="sm"
-            action="secondary"
-            onClick={() => alert("Are you sure you want to cancel stacking?")}
-            width="full"
-          >
-            Cancel Stacking
-          </Button>
+        </ModalFooter>
+      </Modal>
+      <Dialog
+        isOpen={isOpenCancellationDialog}
+        closeAction={() => setIsOpenCancellationDialog(false)}
+      >
+        <DialogContent
+          title=" Are you sure you want to cancel stacking?"
+          description={`The ${remainingFundsDescription}, will be sent to your wallet`}
+        />
+        <DialogFooterActions
+          primaryAction={() => cancelStack()}
+          primaryText="Proceed"
+          secondaryAction={() => setIsOpenCancellationDialog(false)}
+          secondaryText="Cancel"
+        />
+      </Dialog>
+      <DialogConfirmTransactionLoading
+        isOpen={isCancellationProcessing}
+        title={cancellationTx && "Proceeding cancellation"}
+        description={cancellationTx && "Waiting for transaction confirmation."}
+      >
+        {cancellationTx && cancellationTx.hash && (
+          <CancelTransactionLink txHash={cancellationTx.hash} />
+        )}
+      </DialogConfirmTransactionLoading>
+      <Dialog
+        isOpen={isCancellationSuccess}
+        closeAction={() => setIsCancellationSuccess(false)}
+      >
+        <Icon name="check" className="text-primary-400" size={38} />
+        <DialogContent
+          title="Stack Cancelled"
+          description={`The ${remainingFundsDescription}, were sent to your wallet`}
```

Comment:
Just suggesting better punctuation.

```suggestion
          description={`The ${remainingFundsDescription} were sent to your wallet.`}
```